Monika After Story/game/script-affection.rpy

```diff
@@ -2106,6 +2088,163 @@ label monika_affection_nickname_bad_lock:
     m 1efc "Let's talk about something else."
     return
 
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="mas_affection_playernickname",
+            conditional="seen_event('monika_affection_nickname')",
+            action=EV_ACT_QUEUE,
+            aff_range=(mas_aff.AFFECTIONATE, None)
+        )
+    )
+
+default persistent._mas_player_nicknames = list()
+
+label mas_affection_playernickname:
+    python:
+        #A list of names we always want to have
+        base_nicknames = [
+            ("Darling", "darling", True, True, False),
+            ("Honey", "honey", True, True, False),
+            ("Love", "love", True, True, False),
+            ("My love", "my love", True, True, False),
+            ("Sweetheart", "sweetheart", True, True, False),
+            ("Sweetie", "sweetie", True, True, False),
+        ]
+
+    m 1euc "Hey [player]?"
+    m 1eka "Since you can call me by a nickname now, I was wondering if I could call you by a nickname."
+
+    m 1etc "Is that alright with you?{nw}"
+    $ _history_list.pop()
+    menu:
+        m "Is that alright with you?{fast}"
+
+        "Sure, [m_name].":
+            m 1hua "Great!"
+            m 3eud "I should ask though, what names are you comfortable with?"
+            call mas_player_nickname_loop("Diselect the names you're not comfortable with me calling you", base_nicknames)
+
+        "No.":
+            m 1eka "Alright, [player]."
+            m 3eua "Just let me know if you ever change your mind, okay?"
+
+    #Now unlock the nickname change ev
+    $ mas_unlockEVL("monika_change_player_nicknames", "EVE")
+    return "no_unlock"
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_change_player_nicknames",
+            prompt="Can you call me different nicknames?",
+            category=['you'],
+            pool=True,
+            unlocked=False,
+            rules={"no_unlock": None},
+            aff_range=(mas_aff.AFFECTIONATE,None)
+        )
+    )
+
+label monika_change_player_nicknames:
+    m 1hub "Sure [player]!"
+
+    python:
+        #Generate a list of names we're using now so we can set things
+        current_nicknames = [
+            (nickname.capitalize(), nickname, True, True, False)
+            for nickname in persistent._mas_player_nicknames
+        ]
+
+        if not current_nicknames:
+            current_nicknames = [
+                ("Darling", "darling", True, True, False),
+                ("Honey", "honey", True, True, False),
+                ("Love", "love", True, True, False),
+                ("My love", "my love", True, True, False),
+                ("Sweetheart", "sweetheart", True, True, False),
+                ("Sweetie", "sweetie", True, True, False),
```

Comment:
I'd say these should **not** be selected by default. Because if we get to this flow, it means the player disabled them before, right? If you decide to adjust this, don't forget to adjust the dlg line too.